### PR TITLE
connect: close the happy eyeballs loser connection when using QUIC

### PR DESCRIPTION
**Pull request by bagder**

### CODE REVIEW

**Comment by bagder** [comment]
This fixes the larger part of #8915

File: lib/connect.c [hunk 1]
```diff
@@ -910,6 +910,8 @@ CURLcode Curl_is_connected(struct Curl_easy *data,
         conn->tempsock[i] = CURL_SOCKET_BAD;
         post_SOCKS(data, conn, sockindex, connected);
         connkeep(conn, "HTTP/3 default");
+        if(conn->tempsock[other] != CURL_SOCKET_BAD)
+          Curl_quic_disconnect(data, conn, other);
```
Comment (nibanks) [inline lib/connect.c:914]: At least for msh3, I think we need to update `msh3_cleanup` to free the request (if non-null) too, to full fix the problem. Right?

Comment (bagder) [reply lib/connect.c:914]: Yeah, I have a follow-up separate PR for msh3 coming up that fixes the cleanup etc to make sure it behaves correctly and doesn't leak. See #9304 for a more complete take.

Comment (nibanks) [reply lib/connect.c:914]: Ok. Well, I can't approve the PR, but LGTM. 👍🏻

Comment (bagder) [reply lib/connect.c:914]: thanks!

**Pull request closed without merging**